lib/lltz_codegen/lltz_codegen.ml

```diff
@@ -208,76 +216,89 @@ let rec compile : LLTZ.E.t -> t =
     [ (match expr.desc with
        | Variable (Var name) -> compile_variable name
        | Let_in { let_var = Var var; rhs; in_ } -> compile_let_in var rhs in_
-       | Lambda { lam_var = Var var, lam_var_type; return_type; body } ->
-         compile_lambda var lam_var_type return_type body
+       | Lambda { lam_var = Var var, lam_var_type; body } ->
+         compile_lambda expr
        | Lambda_rec
-           { lam_var = Var var, lam_var_type; mu_var = Var mu; return_type; body } ->
-         compile_lambda_rec var lam_var_type mu return_type body
+           { mu_var = Var mu, mu_type; lambda = {lam_var = Var var, lam_var_type; body} } ->
```

Comment:
check warnings

Reply:
it doesn't do it even after supression of warning. I think it is because the variables are necessary in the pattern to match, sa the compiler doesn’t warn and they’re considered "used" in the pattern